app/gui/language/span-tree/src/action.rs

```diff
@@ -286,8 +286,39 @@ impl<'a, T> Implementation for node::Ref<'a, T> {
                         })
                         .map(|found| found.node);
 
+                    // To better understand the code, it's important to know the structure of
+                    // the [`SpanTree`] for prefix chains:
+                    //
+                    // Root
+                    // ├─Chained
+                    // │ ├─Chained
+                    // │ │ ├─Chained
+                    // │ │ │ └─ Argument1
+                    // │ │ └─ Argument2
+                    // │ └─ Argument3
+                    // └─ Argument4
+                    //
+                    // We are iterating over the old span tree, while modifying an expression that
+                    // constantly changes. The assumption here is that as we go, we only modify AST
+                    // tree deeper than we will look in future iterations. That way we can be sure
+                    // that the "outer layers" of AST still corresponds to the original span-tree.
+                    // It is done that way, because we have no way of actually reconstructing the
+                    // span-tree to be entirely correct on every step. The new AST is after all
+                    // wrong and needs to be fixed for a reason. Rebuilding the span-tree before
+                    // replacing named arguments will give you wrong assignments on everything.
+                    // The loop propagates up the old span tree, knowing that the outer layers of
+                    // the AST will still corresponds to it.
```

Comment:
I must admit, I have a strong difficulty in understanding this paragraph:

> We are iterating over the old span tree, while modifying an expression that constantly changes

I suppose the expression is actually an `ast` variable?

> The new AST is after all wrong and needs to be fixed for a reason.

What does it mean, "AST" is wrong? What AST? The old one, or that after applying a change? 

> Rebuilding the span-tree before replacing named arguments will give you wrong assignments on everything.

What is "replacing named arguments"? Looking at the code, I thought we actually are adding new named arguments, or rather: replace the named with the unnamed.

I guess this just to say we change the arguments to named starting from the last one (as it's in the deepest level)? I think such introductory sentence would help me a lot.

Reply:
Updated the paragraph, I hope it is better now.

> I suppose the expression is actually an ast variable?

Yes

> What does it mean, "AST" is wrong? What AST? The old one, or that after applying a change?

Sorry, it was a mistake. The span tree is meant here, not the AST.

> What is "replacing named arguments"?

I removed this sentence. The better word is `rewriting`, though. We rewrite named arguments to a different form.

> I guess this just to say we change the arguments to named starting from the last one (as it's in the deepest level)? I think such introductory sentence would help me a lot.

This is actually the opposite. The last argument is on the top of the tree, see attached ASCII graphics. I added an additional note about it.